scripts/template_matcher.py

```diff
@@ -59,15 +84,58 @@ def _compute_prediction(self, k, img, kp, des):
         kp: keypoints from scene image,   des: descriptors from scene image
         """
 
-        # TODO: find corresponding points in the input image and the templae image
-        #       put keypoints from template image in template_pts
-        #       put corresponding keypoints from input image in img_pts
-
-
-        #TODO: change img to img_T once you do the homography transform
-        visual_diff = compare_images(img, self.signs[k])
+        # Get list of matches between keypoints based on destriptors
+        matches = self.matcher.knnMatch(self.descs[k],des,k=2)
+        # Convert into point form
+        matches = [(m.queryIdx, m.trainIdx) for m, n in matches]
+
+        # Convert those points to points findHomography can handle
```

Comment:
Nice job figuring out the OpenCV API.  The data structures aren't super transparent.